src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -146,6 +146,9 @@ The .NET Foundation licenses this file to you under the MIT license.
       <NetCoreAppNativeLibrary Include="icuuc" Condition="'$(_IsiOSLikePlatform)' == 'true' and '$(InvariantGlobalization)' != 'true' and '$(HybridGlobalization)' != 'true'" />
     </ItemGroup>
 
+    <!-- zlib-ng's libz.a is installed in the aotsdk folder but needs to be consumed as a NetCoreAppNativeLibrary, so it needs to change location to be detected properly when escaping it. -->
```

Comment:
Why can’t you place this library into the right folder instead?

Reply:
Can you help me confirm I found the correct location?:

[this](https://github.com/dotnet/runtime/pull/104454/files#diff-5cd57a2173307aaec58099abe5dc12175d1e5aa72a35c2a2b283a735d8af158cR187) is the installation line I added for the zlib library:

```cmake
if((NOT CLR_CMAKE_USE_SYSTEM_ZLIB) AND STATIC_LIBS_ONLY)
    install_static_library(zlib aotsdk nativeaot)
endif()
```
I assume the `aotsdk` argument is what needs to change. Is this correct?

If yes, then I am unsure which of the different System.IO.Compression.Native install lines contains the correct directory to use for zlib. These are the ones I see that apply to Linux:

https://github.com/carlossanlop/runtime/blob/d70f22cf1dd398b987c7db0653a5529024bc0568/src/native/libs/System.IO.Compression.Native/CMakeLists.txt#L97

```cmake
if (CLR_CMAKE_TARGET_UNIX OR CLR_CMAKE_TARGET_BROWSER OR CLR_CMAKE_TARGET_WASI)
    ...
    if (GEN_SHARED_LIB)
        ...
        install_with_stripped_symbols (System.IO.Compression.Native PROGRAMS .)
```


Or the one at the end of the file:

https://github.com/carlossanlop/runtime/blob/d70f22cf1dd398b987c7db0653a5529024bc0568/src/native/libs/System.IO.Compression.Native/CMakeLists.txt#L190
```cmake
install (TARGETS System.IO.Compression.Native-Static DESTINATION ${STATIC_LIB_DESTINATION} COMPONENT libs)
```

I see that `${STATIC_LIB_DESTINATION}`, is defined as:

https://github.com/carlossanlop/runtime/blob/6b03ebd5c3fc8fe851a533d1044cb17455ad4971/src/native/libs/CMakeLists.txt
```cmake
if (STATIC_LIBS_ONLY)
    ...
    set(STATIC_LIB_DESTINATION lib)
else ()
    ...
    set(STATIC_LIB_DESTINATION .)
endif ()
```

And I think this refers to the folder `artifacts/bin/coreclr/lib`, which contains a copy of `libSystem.IO.Compression.Native.a`. Although I don't see a copy of `icudata` there as it was mentioned in the chat.

So is `STATIC_LIB_DESTINATION` the right folder?

cc @hoyosjs, I was discussing this with him earlier.

Reply:
I guess I'll try it 🙂 I'll let you know if I break everything.

Reply:
Yeah, copying with MSBuild won't do it, the CI is currently failing:

```
/__w/1/s/artifacts/bin/coreclr/linux.x64.Release/build/Microsoft.NETCore.Native.Unix.targets(150,5): error MSB3026: Could not copy "/__w/1/s/artifacts/bin/coreclr/linux.x64.Release/aotsdk/libz.a" to "/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.linux-x64/Release/runtimes/linux-x64/native/libz.a". Beginning retry 1 in 1000ms. The process cannot access the file '/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.linux-x64/Release/runtimes/linux-x64/native/libz.a' because it is being used by another process.  [/__w/1/s/artifacts/bin/aotTests/projects/System.ComponentModel.TypeConverter.TrimmingTests/TypeDescriptorTest_FeatureSwitch_Unregistered/linux-x64/project.csproj]
```